app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt

```diff
@@ -45,3 +45,26 @@ class IndianPhoneNumberValidator : PhoneNumberValidator {
     }
   }
 }
+
+class EthiopianPhoneNumberValidator : PhoneNumberValidator {
+  override fun validate(number: String, type: PhoneNumberValidator.Type): Result {
+    return when(type) {
+      LANDLINE_OR_MOBILE -> {
+        when{
+          number.isBlank() -> Result.BLANK
+          number.length < 9 -> LENGTH_TOO_SHORT
+          number.length > 10 -> LENGTH_TOO_LONG
+          else -> VALID
+        }
+      }
+      MOBILE -> {
+        when{
+          number.isBlank() -> Result.BLANK
+          number.length < 9 -> LENGTH_TOO_SHORT
+          number.length > 10 -> LENGTH_TOO_LONG
+          else -> VALID
+        }
+      }
+    }
```

Comment:
Since these two code blocks are identical, we can skip the check of the `type` and just add the validations directly, right?

Reply:
Made this change in this commit `Extract validation logic into discrete methods in `PhoneNumberValidator``. Since we are already separating it into different functions.